### PR TITLE
Fix incorrect emit for JSX/React

### PR DESCRIPTION
**Pull request by rbuckton**
This fixes some issues in the jsx transform and pretty printer that were causing a significant number of tests to fail revolving around JSX and React emit.

CC: @vladima, @DanielRosenwasser, @yuit

### CODE REVIEW

**Comment by vladima** [comment]
does this incur any changes in test baselines?

**Comment by rbuckton** [comment]
No, it fixes the emit to match existing baselines.

**Comment by vladima** [comment]
:+1:

**Comment by RyanCavanaugh** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
:ship: :it:

File: src/compiler/printer.ts [hunk 1]
```diff
@@ -528,12 +528,16 @@ const _super = (function (geti, seti) {
                     // JSX (non-expression)
```
Comment (yuit) [inline src/compiler/printer.ts]: Minor: should this comment be updated? Since the SyntaxKind that it will print is `JsxExpression`

Comment (rbuckton) [inline src/compiler/printer.ts]: JsxExpression is not an expression, you can't put it in a CommaExpression for example. Rather its more like ExpressionStatement, it represents a different part of the tree that contains an expression. The fact that it inherits from `Expression` is a red-herring and one of the reasons I wrote this incorrectly in the first place.

**Pull request merged**